File: Medium/2161. Partition Array According to Given Pivot.py
from typing import List


class TwoPointers:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        n = len(nums)
        res = [pivot] * n
        left, right = 0, n - 1
        for i in range(n):
            if nums[i] < pivot:
                res[left] = nums[i]
                left += 1
        for j in range(n - 1, -1, -1):
            if nums[j] > pivot:
                res[right] = nums[j]
                right -= 1
        return res


class BruteForce:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        left, mid, right = [], [], []
        for num in nums:
            if num > pivot:
                right.append(num)
            elif num < pivot:
                left.append(num)
            else:
                mid.append(num)
        return left + mid + right

File: Medium/2161. Partition Array According to Given Pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int n = nums.length;
        int[] res = new int[n];
        Arrays.fill(res, pivot);
        int l = 0;
        for (int num : nums) {
            if (num < pivot) {
                res[l++] = num;
            }
        }
        int r = n - 1;
        for (int i = n - 1; i > -1; i--) {
            if (nums[i] > pivot) {
                res[r--] = nums[i];
            }
        }
        return res;
    }
}

class Solution1 {
    public int[] pivotArray(int[] nums, int pivot) {
        List<Integer> smaller = new ArrayList<>();
        List<Integer> bigger = new ArrayList<>();
        int same = 0;
        for (int num : nums) {
            if (num < pivot) {
                smaller.add(num);
            } else if (num > pivot) {
                bigger.add(num);
            } else {
                same++;
            }
        }
        int[] res = new int[nums.length];
        int i = 0;
        for (int num : smaller) {
            res[i++] = num;
        }
        for (int j = 0; j < same; j++) {
            res[i++] = pivot;
        }
        for (int num : bigger) {
            res[i++] = num;
        }
        return res;
    }
}
